Remove all Bazel invocations from test-e2e-cip

tylerferrara opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

- If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
- Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
- If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
- If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

#### What type of PR is this?
/kind cleanup
/kind deprecation
<!--
Add one of the following kinds:
/kind bug
/kind cleanup
/kind documentation
/kind feature
/kind design

Optionally add one or more of the following kinds if applicable:
/kind api-change
/kind deprecation
/kind failing-test
/kind flake
/kind regression
-->

#### What this PR does / why we need it:
This change removes all invocations of `bazel run` in favor of `go run` within e2e.go. In addition, the `test-e2e-cip` make target, which executes e2e.go, no longer relies on Bazel.

Partially satisfies #304 

#### Which issue(s) this PR fixes:

<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.

Fixes #

or

None
-->

#### Special notes for your reviewer:

#### Does this PR introduce a user-facing change?

<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->

```release-note
Replace bazel invocations, inside e2e.go, with go run command.
Remove Bazel invocation from make target: test-e2e-cip
```

tylerferrara commented:
Bazel automatically stamps go binaries with the following information:
- GitDescribe
- GitCommit
- TimestampUtcRfc3339

Go can achieve this same functionality by specifying `-ldflags` during build or runtime. Thorefore, the Makefile now stamps e2e.go with the same information to output the following log information:
```
I0621 15:31:17.918689   32131 e2e.go:325] Built:   2021-06-21 19:31:17+0000
I0621 15:31:17.918747   32131 e2e.go:326] Version: v1.337.0-12-gc60e70e-dirty
I0621 15:31:17.918750   32131 e2e.go:327] Commit:  c60e70efae2213238e41267c17176cf9f273cddf
```

tylerferrara commented:
cc: @listx @justaugustus @amwat
/assign @justaugustus

justaugustus reviewed (changes requested):
@tylerferrara -- There are two files to remove (and perhaps add entries in .gitignore for).

Non-blocking (can be done in a follow-up):
Consider matching the logic in https://github.com/kubernetes/release/blob/99026391e1e90872224afd2de5903d51551c4b07/compile-release-tools#L67-L88

```bash
compile_with_flags() {
  local git_tree_state
  local pkg
  local tool="$1"

  git_tree_state=dirty
  pkg=k8s.io/release/pkg/version

  if git_status=$(git status --porcelain --untracked=no 2>/dev/null) && [[ -z "${git_status}" ]]; then
    git_tree_state=clean
  fi

  go build -v -ldflags "-s -w \
    -X $pkg.buildDate=$(date -u +'%Y-%m-%dT%H:%M:%SZ') \
    -X $pkg.gitCommit=$(git rev-parse HEAD 2>/dev/null || echo unknown) \
    -X $pkg.gitTreeState=$git_tree_state \
    -X $pkg.gitVersion=$(git describe --tags --abbrev=0 || echo unknown)" \
    -o "$RELEASE_TOOL_BIN/$tool" "./cmd/$tool" \
    || return 1

  echo "$tool was successfully compiled and installed to $RELEASE_TOOL_BIN/$tool"
}
```

using the `internal/version` package that was pulled in https://github.com/kubernetes-sigs/k8s-container-image-promoter/pull/294 to further clean up e2e.go.

tylerferrara commented:
> @tylerferrara -- There are two files to remove (and perhaps add entries in .gitignore for).
> 
> Non-blocking (can be done in a follow-up):
> Consider matching the logic in https://github.com/kubernetes/release/blob/99026391e1e90872224afd2de5903d51551c4b07/compile-release-tools#L67-L88
> 
> ```shell
> compile_with_flags() {
>   local git_tree_state
>   local pkg
>   local tool="$1"
> 
>   git_tree_state=dirty
>   pkg=k8s.io/release/pkg/version
> 
>   if git_status=$(git status --porcelain --untracked=no 2>/dev/null) && [[ -z "${git_status}" ]]; then
>     git_tree_state=clean
>   fi
> 
>   go build -v -ldflags "-s -w \
>     -X $pkg.buildDate=$(date -u +'%Y-%m-%dT%H:%M:%SZ') \
>     -X $pkg.gitCommit=$(git rev-parse HEAD 2>/dev/null || echo unknown) \
>     -X $pkg.gitTreeState=$git_tree_state \
>     -X $pkg.gitVersion=$(git describe --tags --abbrev=0 || echo unknown)" \
>     -o "$RELEASE_TOOL_BIN/$tool" "./cmd/$tool" \
>     || return 1
> 
>   echo "$tool was successfully compiled and installed to $RELEASE_TOOL_BIN/$tool"
> }
> ```
> 
> using the `internal/version` package that was pulled in #294 to further clean up e2e.go.

Added new `stamp_ldflags.sh` which does the same thing as k/release. Also updated previous comment to compare previous and current logging info.

listx commented on Makefile:
```diff
@@ -17,14 +17,19 @@
 
+# Linker Flags
+TIMESTAMP_UTC_RFC3339:=$(shell date -u +"%Y-%m-%d %H:%M:%S%z")
+STABLE_GIT_DESC:=$(shell git describe --always --dirty --long)
```
Remove, because we're now using the `version` package, right?

listx commented on stamp_ldflags.sh:
```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env bash
```
Rename to `go_wrapped` or some other `go_*` ?

justaugustus replied:
Let's do `go_with_version.sh`.

justaugustus reviewed (approved):
Great work, @tylerferrara!

listx commented:
/lgtm
/approve
